tests/test_data_processor.py

```diff
@@ -238,48 +348,7 @@ def test_first_update(self, data_processor, data_packets, init_alt, max_alt):
         assert len(processed_data) == len(data_packets)
         for idx, data in enumerate(processed_data):
             assert data.current_altitude == d._current_altitudes[idx]
-            assert data.velocity == d._vertical_velocities[idx]
-
-    def test_previous_velocity_retained(self, data_processor):
-        """Test that previous velocity is retained correctly between updates."""
-        # Initial data packet
-        x_accel = [1, 2, 3]
-        y_accel = [0, 0, 0]
-        z_accel = [0, 0, 0]
-
-        packets = [
-            EstimatedDataPacket(
-                1.0 * 1e9,
-                estLinearAccelX=x_accel[0],
-                estLinearAccelY=y_accel[0],
-                estLinearAccelZ=z_accel[0],
-                estPressureAlt=0,
-            ),
-            EstimatedDataPacket(
-                2.0 * 1e9,
-                estLinearAccelX=x_accel[1],
-                estLinearAccelY=y_accel[1],
-                estLinearAccelZ=z_accel[1],
-                estPressureAlt=0,
-            ),
-        ]
-        data_processor.update(packets)
-        first_vel = data_processor._previous_velocity
-
-        # Additional data packet
-        new_packets = [
-            EstimatedDataPacket(
-                3.0 * 1e9,
-                estLinearAccelX=x_accel[2],
-                estLinearAccelY=y_accel[2],
-                estLinearAccelZ=z_accel[2],
-                estPressureAlt=0,
-            ),
-        ]
-        data_processor.update(new_packets)
-        second_vel = data_processor._previous_velocity
-
-        assert second_vel[0] > first_vel[0], "Previous velocity should be updated after each data update."
+            assert data.vertical_velocity == d._vertical_velocities[idx]
```

Comment:
please add more tests here, checking the new fields vertical_acceleration, and time_since_last_data_packet